tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,180 @@
+        rnd_seed = time.time()
+        rnd_seed_override = test_ctx.globals.get("random_seed", True)
+        if not rnd_seed_override is None:
+            rnd_seed = rnd_seed_override
+        self.logger.info(f"Random seed: {rnd_seed}")
+        self.rnd = random.Random(rnd_seed)
+        self.rpk = RpkTool(self.redpanda)
+
+    class ConfigProp(Enum):
+        QUOTA_NODE_MAX_IN = "kafka_throughput_limit_node_in_bps"
+        QUOTA_NODE_MAX_EG = "kafka_throughput_limit_node_out_bps"
+        THROTTLE_DELAY_MAX_MS = "max_kafka_throttle_delay_ms"
+        BAL_WINDOW_MS = "kafka_quota_balancer_window_ms"
+        BAL_PERIOD_MS = "kafka_quota_balancer_node_period_ms"
+        QUOTA_SHARD_MIN_RATIO = "kafka_quota_balancer_min_shard_thoughput_ratio"
+        QUOTA_SHARD_MIN_BPS = "kafka_quota_balancer_min_shard_thoughput_bps"
+
+    def get_config_parameter_random_value(self, prop: ConfigProp):
+        if prop in [
+                self.ConfigProp.QUOTA_NODE_MAX_IN,
+                self.ConfigProp.QUOTA_NODE_MAX_EG
+        ]:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return None
+            if r == 1:
+                return 32  # practical minimum
+            return math.floor(2**(self.rnd.random() * 36 + 4))  # up to 1 TB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_BPS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 30))  # up to 1 GB/s
+
+        if prop == self.ConfigProp.QUOTA_SHARD_MIN_RATIO:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            if r == 1:
+                return 1
+            return self.rnd.random()
+
+        if prop == self.ConfigProp.THROTTLE_DELAY_MAX_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 25))  # up to ~1 year
+
+        if prop == self.ConfigProp.BAL_WINDOW_MS:
+            r = self.rnd.randrange(4)
+            if r == 0:
+                return 1
+            if r == 1:
+                return 2147483647
+            return math.floor(2**(self.rnd.random() * 31))
+
+        if prop == self.ConfigProp.BAL_PERIOD_MS:
+            r = self.rnd.randrange(3)
+            if r == 0:
+                return 0
+            return math.floor(2**(self.rnd.random() * 22))  # up to ~1.5 months
+
+        raise Exception(f"Unsupported ConfigProp: {prop}")
+
+    def setUp(self):
+
+        self.config = {}
+        for prop in list(self.ConfigProp):
+            val = self.get_config_parameter_random_value(prop)
+            self.config[prop] = val
+            if not val is None:
+                self.redpanda.add_extra_rp_conf({prop.value: val})
+
+        self.logger.info(
+            f"!SETTINGUP RP! extras:{self.redpanda._extra_rp_conf}")
+        super(ThroughputLimitsSnc, self).setUp()
```

Comment:
nit: looks like you could also do this in the __init__ constructor. you can pass a dictionary of rp configs for the new cluster that is going to be created each test.

Reply:
that probably won't gain us much advantage, because if I run multiple tests of this class they still share the same generator, and even though the initial configuration is the same for them, the ultimate test results still depend on the list and the order of the tests executed

Reply:
is this true? i always believed that a new instance of the entire class was created for each test. but i've never tried to verify that.

Reply:
i just ran a test with

```diff
+class CtorTest(RedpandaTest):
+    def __init__(self, test_context):
+        self.r = random.randint(0, 100)
+        super(CtorTest, self).__init__(test_context=test_context)
+
+    @cluster(num_nodes=3)
+    def test_abc(self):
+        print("XXXXXXXXXXX", self.r)
+
+    @cluster(num_nodes=3)
+    def test_def(self):
+        print("XXXXXXXXXXX", self.r)
+
```

i see each instance getting a different value for self.r.  so i think you can safely move your setup into the constructor and populate the redpanda configuration there?